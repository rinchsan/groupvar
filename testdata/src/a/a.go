package a

var _ int
var _, _, _ int
var _, _, _, _ int // want "grouped var declarations count should be up to 3"

var (
	_       int
	_, _    int
	_, _, _ int
)
var (
	_, _, _    int
	_, _, _, _ int // want "grouped var declarations count should be up to 3"
)

var _ = 1
var _, _, _ = 1, 2, 3
var _, _, _ = "foo", "bar", "baz"
var _, _, _, _ = 1, 2, 3, 4                    // want "grouped var declarations count should be up to 3"
var _, _, _, _ = "foo", "bar", "baz", "foobar" // want "grouped var declarations count should be up to 3"
var _, _, _, _ = 1, "foo", 2, "bar"            // want "grouped var declarations count should be up to 3"
var _, _ = 1, "foo"                            // want "grouped var declarations should be separated by types"

const _ = 1
const _, _, _ = 1, 2, 3
const _, _, _ = "foo", "bar", "baz"
const _, _, _, _ = 1, 2, 3, 4                    // want "grouped const declarations count should be up to 3"
const _, _, _, _ = "foo", "bar", "baz", "foobar" // want "grouped const declarations count should be up to 3"
const _, _, _, _ = 1, "foo", 2, "bar"            // want "grouped const declarations count should be up to 3"
const _, _ = 1, "foo"                            // want "grouped const declarations should be separated by types"

var (
	_, _, _ = 1, 2, 3
	_, _, _ = "foo", "bar", "baz"
)
var (
	_, _, _, _ = 1, 2, 3, 4                    // want "grouped var declarations count should be up to 3"
	_, _, _, _ = "foo", "bar", "baz", "foobar" // want "grouped var declarations count should be up to 3"
	_, _       = 1, "foo"                      // want "grouped var declarations should be separated by types"
)

const (
	_, _, _ = 1, 2, 3
	_, _, _ = "foo", "bar", "baz"
)
const (
	_, _, _, _ = 1, 2, 3, 4                    // want "grouped const declarations count should be up to 3"
	_, _, _, _ = "foo", "bar", "baz", "foobar" // want "grouped const declarations count should be up to 3"
	_, _       = 1, "foo"                      // want "grouped const declarations should be separated by types"
)

func f() {
	var _ int
	var _, _, _ int
	var _, _, _, _ int // want "grouped var declarations count should be up to 3"

	var (
		_       int
		_, _    int
		_, _, _ int
	)
	var (
		_, _, _    int
		_, _, _, _ int // want "grouped var declarations count should be up to 3"
	)

	var _ = 1
	var _, _, _ = 1, 2, 3
	var _, _, _ = "foo", "bar", "baz"
	var _, _, _, _ = 1, 2, 3, 4                    // want "grouped var declarations count should be up to 3"
	var _, _, _, _ = "foo", "bar", "baz", "foobar" // want "grouped var declarations count should be up to 3"
	var _, _, _, _ = 1, "foo", 2, "bar"            // want "grouped var declarations count should be up to 3"
	var _, _ = 1, "foo"                            // want "grouped var declarations should be separated by types"

	var (
		_, _, _ = 1, 2, 3
		_, _, _ = "foo", "bar", "baz"
	)
	var (
		_, _, _, _ = 1, 2, 3, 4                    // want "grouped var declarations count should be up to 3"
		_, _, _, _ = "foo", "bar", "baz", "foobar" // want "grouped var declarations count should be up to 3"
		_, _, _, _ = 1, "foo", 2, "bar"            // want "grouped var declarations count should be up to 3"
		_, _       = 1, "foo"                      // want "grouped var declarations should be separated by types"
	)
}
