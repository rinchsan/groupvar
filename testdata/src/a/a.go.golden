package a

var _ int
var _, _, _ int
var (
	_ int
	_ int
	_ int
	_ int
) // want "grouped var declarations count should be up to 3"

var (
	_       int
	_, _    int
	_, _, _ int
)
var (
	_, _, _ int
	_       int
	_       int
	_       int
	_       int // want "grouped var declarations count should be up to 3"
)

var _ = 1
var _, _, _ = 1, 2, 3
var _, _, _ = "foo", "bar", "baz"
var (
	_ = 1
	_ = 2
	_ = 3
	_ = 4
) // want "grouped var declarations count should be up to 3"
var (
	_ = "foo"
	_ = "bar"
	_ = "baz"
	_ = "foobar"
) // want "grouped var declarations count should be up to 3"
var (
	_ = 1
	_ = "foo"
	_ = 2
	_ = "bar"
) // want "grouped var declarations count should be up to 3"
var (
	_ = 1
	_ = "foo"
) // want "grouped var declarations should be separated by types"

const _ = 1
const _, _, _ = 1, 2, 3
const _, _, _ = "foo", "bar", "baz"
const (
	_ = 1
	_ = 2
	_ = 3
	_ = 4
) // want "grouped const declarations count should be up to 3"
const (
	_ = "foo"
	_ = "bar"
	_ = "baz"
	_ = "foobar"
) // want "grouped const declarations count should be up to 3"
const (
	_ = 1
	_ = "foo"
	_ = 2
	_ = "bar"
) // want "grouped const declarations count should be up to 3"
const (
	_ = 1
	_ = "foo"
) // want "grouped const declarations should be separated by types"

var (
	_, _, _ = 1, 2, 3
	_, _, _ = "foo", "bar", "baz"
)
var (
	_ = 1
	_ = 2
	_ = 3
	_ = 4 // want "grouped var declarations count should be up to 3"
	_ = "foo"
	_ = "bar"
	_ = "baz"
	_ = "foobar" // want "grouped var declarations count should be up to 3"
	_ = 1
	_ = "foo" // want "grouped var declarations should be separated by types"
)

const (
	_, _, _ = 1, 2, 3
	_, _, _ = "foo", "bar", "baz"
)
const (
	_ = 1
	_ = 2
	_ = 3
	_ = 4 // want "grouped const declarations count should be up to 3"
	_ = "foo"
	_ = "bar"
	_ = "baz"
	_ = "foobar" // want "grouped const declarations count should be up to 3"
	_ = 1
	_ = "foo" // want "grouped const declarations should be separated by types"
)

func f() {
	var _ int
	var _, _, _ int
	var (
		_ int
		_ int
		_ int
		_ int
	) // want "grouped var declarations count should be up to 3"

	var (
		_       int
		_, _    int
		_, _, _ int
	)
	var (
		_, _, _ int
		_       int
		_       int
		_       int
		_       int // want "grouped var declarations count should be up to 3"
	)

	var _ = 1
	var _, _, _ = 1, 2, 3
	var _, _, _ = "foo", "bar", "baz"
	var (
		_ = 1
		_ = 2
		_ = 3
		_ = 4
	) // want "grouped var declarations count should be up to 3"
	var (
		_ = "foo"
		_ = "bar"
		_ = "baz"
		_ = "foobar"
	) // want "grouped var declarations count should be up to 3"
	var (
		_ = 1
		_ = "foo"
		_ = 2
		_ = "bar"
	) // want "grouped var declarations count should be up to 3"
	var (
		_ = 1
		_ = "foo"
	) // want "grouped var declarations should be separated by types"

	var (
		_, _, _ = 1, 2, 3
		_, _, _ = "foo", "bar", "baz"
	)
	var (
		_ = 1
		_ = 2
		_ = 3
		_ = 4 // want "grouped var declarations count should be up to 3"
		_ = "foo"
		_ = "bar"
		_ = "baz"
		_ = "foobar" // want "grouped var declarations count should be up to 3"
		_ = 1
		_ = "foo"
		_ = 2
		_ = "bar" // want "grouped var declarations count should be up to 3"
		_ = 1
		_ = "foo" // want "grouped var declarations should be separated by types"
	)
}
